src/main/java/com/uspray/uspray/service/ShareFacade.java

```diff
@@ -52,22 +52,46 @@ public List<SharedPrayResponseDto> getSharedPrayList(String username) {
     }
 
     @Transactional
-    public void receivedSharedPray(String username, SharedPrayRequestDto sharedPrayRequestDto) {
+    public Long receivedSharedPray(String username, SharedPrayRequestDto sharedPrayRequestDto) {
 
         Member member = memberRepository.getMemberByUserId(username);
         List<Pray> prayList = prayRepository.findAllByIdIn(sharedPrayRequestDto.getPrayIds());
+        long total = 0L;
 
         if (prayList.size() != sharedPrayRequestDto.getPrayIds().size()) {
             throw new NotFoundException(ErrorStatus.PRAY_NOT_FOUND_EXCEPTION,
                 ErrorStatus.PRAY_NOT_FOUND_EXCEPTION.getMessage());
         }
         for (Pray pray : prayList) {
+            if (!receiveCheck(member, pray)) {
+                continue;
+            }
             SharedPray sharedPray = SharedPray.builder()
                 .member(member)
                 .pray(pray)
                 .build();
             sharedPrayRepository.save(sharedPray);
+            total++;
         }
+
+        if (total == 0L) {
+            throw new CustomException(ErrorStatus.CANNOT_RECEIVE_SHARED_PRAY_EXCEPTION,
+                ErrorStatus.CANNOT_RECEIVE_SHARED_PRAY_EXCEPTION.getMessage());
+        }
```

Comment:
total 값이 어디서 따로 쓰이지 않는다면 
```suggestion
            if (!receiveCheck(member, pray)) {
                throw new CustomException(ErrorStatus.CANNOT_RECEIVE_SHARED_PRAY_EXCEPTION,
                ErrorStatus.CANNOT_RECEIVE_SHARED_PRAY_EXCEPTION.getMessage());
            }
            SharedPray sharedPray = SharedPray.builder()
                .member(member)
                .pray(pray)
                .build();
            sharedPrayRepository.save(sharedPray);
            total++;
        }
```
이런식으로 바로 예외 터트리는 것은 어떤가요??